lib/modules/platform/bitbucket/schema.ts

```diff
@@ -20,3 +22,58 @@ const PagedSchema = z.object({
 export const PagedSourceResultsSchema = PagedSchema.extend({
   values: z.array(SourceResultsSchema),
 });
+
+export const RepoInfo = z
+  .object({
+    parent: z.unknown().optional().catch(undefined),
+    mainbranch: z.object({
+      name: z.string(),
+    }),
+    has_issues: z.boolean().catch(() => {
+      logger.once.warn('Bitbucket: "has_issues" field missing from repo info');
+      return false;
+    }),
+    uuid: z.string(),
+    full_name: z
+      .string()
+      .regex(
+        /^[^/]+\/[^/]+$/,
+        'Expected repository full_name to be in the format "owner/repo"',
+      ),
+    is_private: z.boolean().catch(() => {
+      logger.once.warn('Bitbucket: "is_private" field missing from repo info');
+      return true;
+    }),
+    project: z
+      .object({
+        name: z.string(),
+      })
+      .nullable()
+      .catch(() => {
+        logger.once.warn('Bitbucket: "project" field missing from repo info');
```

Comment:
Does every repo need to belong to a project?

Reply:
I have no idea, eventually we'll see from logs

Reply:
Do we treat it as mandatory today?

Reply:
Yes

Reply:
I'm not sure we should warn here. Today a missing project would only affect when autodiscoverProjects is configured, wouldn't it? And that's a new feature

Reply:
Let's see all the warnings of these types, and if there is no any, it means we could restrict all the fields

Reply:
I prefer to remove the warn here. It doesn't matter to us if there's no project